hermes/test/test_api.py

```diff
@@ -666,13 +670,28 @@ def test_good_spectroscopy_section_submits_ok(self):
         self.assertEqual(result.status_code, 200)
         self.assertEqual(result.json(), {})
 
-    def test_spectroscopy_requires_flux_and_wavelength(self):
+    def test_spectroscopy_requires_flux_and_wavelength_or_file(self):
         bad_message = deepcopy(self.good_message)
         del bad_message['data']['spectroscopy'][0]['flux']
         del bad_message['data']['spectroscopy'][0]['wavelength']
         result = self.client.post(reverse('submit_message-validate'), bad_message, content_type="application/json")
-        self.assertContains(result, 'flux', status_code=200)
-        self.assertContains(result, 'wavelength', status_code=200)
+        self.assertContains(result, 'Must specify a spectroscopy file to upload or specify one or more flux values',
+                            status_code=200)
+
+    def test_spectroscopy_accepts_files(self):
+        good_message = deepcopy(self.good_message)
+        del good_message['data']['spectroscopy'][0]['flux']
+        del good_message['data']['spectroscopy'][0]['wavelength']
+        good_message['data']['spectroscopy'][0]['files'] = [
+            {
+                'name': 'MyFile1.fits',
```

Comment:
Do we validate that there is some way to pull flux out of the file?

Reply:
no... If  you want to restrict by what is in the file, I need some rules/example files to know what I am restricting based on. The easiest would be to restrict by file type, i.e. allow .fits and .csv and .txt or something, but if you want us to actually check if the .fits or .csv has flux value, we need some rules defined on what constitutes a file with flux values...